src/test/scala/io/iohk/ethereum/jsonrpc/EthJsonRpcControllerSpec.scala

```diff
@@ -0,0 +1,228 @@
+package io.iohk.ethereum.jsonrpc
+
+import io.iohk.ethereum.Fixtures
+import io.iohk.ethereum.db.components.{SharedEphemDataSources, Storages}
+import io.iohk.ethereum.domain.{Block, BlockchainImpl}
+import io.iohk.ethereum.jsonrpc.JsonSerializers.{BlockResponseSerializer, QuantitiesSerializer}
+import io.iohk.ethereum.network.p2p.messages.PV62.BlockBody
+import org.json4s.{DefaultFormats, Extraction}
+import org.json4s.JsonAST._
+import org.scalatest.{Matchers, WordSpec}
+import org.scalatest.prop.PropertyChecks
+import org.spongycastle.util.encoders.Hex
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+
+class EthJsonRpcControllerSpec extends WordSpec with Matchers with PropertyChecks {
```

Comment:
> The service is handled by the routes

I would say 2. and 3. are the same thing because a route is just a method name in the JSON request. And this is covered by `JsonRpcControllerSpec`. Unless you think we also need handled some corner cases *for each method* when encoding/decoding.

Anyway, it seems the testing pattern I proposed was also assumed by @LukasGasior1 in his #178. Lukasz, what's your opinion on this?